sourced/ml/core/algorithms/token_parser.py

```diff
@@ -49,6 +54,9 @@ def __init__(self, stem_threshold=STEM_THRESHOLD, max_token_length=MAX_TOKEN_LEN
         self._single_shot = single_shot
         self._save_token_style = save_token_style
         self._attach_upper = attach_upper
+        self._nn_model = nn_model
+        self._use_nn = use_nn
```

Comment:
```suggestion
```
because we can always replace this with `self._id_splitter_nn is not None`

Here the check is needed, we can write `if use_nn:`